vue/src/components/MapLibre.vue

```diff
@@ -151,50 +152,73 @@ watch([() => state.timestamp, () => state.filters, () => state.satellite, () =>
 
 });
 
-watch(
-  () => state.bbox,
-  (val) => fitBounds(val)
-);
-
-watch(() => props.compact, () => {
-  // Wait for it to resize the map based on CSS before resizing and fitting
-  nextTick(() => {
-    map.value?.resize();
-    if (props.compact) {
-      fitBounds(state.bbox);
-    }
+let loadedLocalLayerIds: number[] = [];
+
+watch([shallowRef(map), () => state.localMapFeatureIds], ([, newIds]) => {
+  if (!map.value) return;
+
+  const addedIds = newIds.filter((id) => !loadedLocalLayerIds.includes(id));
+  const removedIds = loadedLocalLayerIds.filter((id) => !newIds.includes(id));
+
+  loadedLocalLayerIds = [...newIds];
+
+  removedIds.forEach((id) => {
+    map.value?.removeLayer(`local-layer-${id}:fill`);
+    map.value?.removeLayer(`local-layer-${id}:outline`);
+    map.value?.removeSource(`local-source-${id}`);
```

Comment:
This was something done before my time with the project, but toggling different modelRuns basically rebuilds the entire data passed to the model.  That means that switching tabs to the model runs and turning on any model run will make the local geoJSON features disappear.  It may be necessary to add some data to the the sources in the main style function:
https://github.com/ResonantGeoData/RD-WATCH/blob/main/vue/src/mapstyle/index.ts#L34

Not the way I would do stuff if starting from the beginning (I'd rather use maplibre's addSource/addLayer) but you probably need to add an additional function here to make sure the localGeoJSONs don't get disabled when running.

Reply:
I fixed the issue and moved the layer drawing into the `style` function.